runtime/doc/api.txt

```diff
@@ -2649,8 +2649,14 @@ nvim_buf_set_extmark({buffer}, {ns_id}, {line}, {col}, {opts})
                     lines are placed below the buffer line containing the
                     mark.
                   • virt_lines_above: place virtual lines above instead.
-                  • virt_lines_leftcol: Place virtual lines in the leftmost
-                    column of the window, bypassing sign and number columns.
+                  • virt_lines_leftcol: Can be a boolean or the string
+                    "sidescroll".
+                    • `false`: virtual lines start after sign and number
+                      columns, etc..
+                    • `true`: virtual lines start at the leftmost column of
+                      the window.
+                    • "sidescroll": with 'nowrap' virtual lines can be
+                      scrolled horizontally, otherwise the same as `false`.
```

Comment:
Why don't we make `false` work like this?

Reply:
See my last comment in #31000

Reply:
Do you mean like how `spell` currently works? That may be a bit confusing, since `leftcol` can also refer to the current horizontal offset of a window, so one would expect `virt_lines_leftcol = false` to disable horizontal scrolling.

Reply:
Maybe allowing horizontal scrolling if and only if `virt_lines_leftcol` is omitted make more sense?

Reply:
I've updated this PR following <https://github.com/neovim/neovim/issues/18282#issuecomment-2667771187>. `virt_lines_overflow` works independently from `virt_lines_leftcol`.